### DISCUSSION TITLE
Twig langusty firmatter

### DISCUSSION THREAD

**Discussion by slawekmoskal** [microsoft/vscode-discussions#521] [Extension Development QnA] [answered]
Hi. Vscode is great except for one thing. does not support the twig language. I can't deal with formatting twig files that have java script in addition. I've tried all plugins and still nothing. do any of you use twig and vscode?

**Comment by alefragnani** [comment, marked as answer]
Hi @slawekmoskal ,

I don’t know/use Twig language, but a quick look in the marketplace shows a lot of extensions. These two (https://marketplace.visualstudio.com/items?itemName=junstyle.twig-language  and https://marketplace.visualstudio.com/items?itemName=artagnon.vstwig, in particular, seems to have a _formatter_, so I suggest you to try out.

If these are not the extensions you use, or their formatter does not fit your need, the best location to ask for a fix/update, is in those extension’s repo, or the extension repo you use for twig syntax highlight, as VS Code does not provide built in twig support.

Hope this helps
